BeardedManStudios/Source/BMSByte.cs

```diff
@@ -380,150 +360,24 @@ public void Clear(bool raw = false)
 			if (raw)
 				Append(new byte[] { 1 });
 		}
-
+		
 		/// <summary>
 		/// Get an array of a specific type from the internally tracked byte array
 		/// </summary>
 		/// <param name="type">The type of array to be pulled</param>
 		/// <param name="o"></param>
 		/// <returns>The updated array of data</returns>
-		private Array GetArray(Type type, object o)
+		private Array GetArray<T>(object o)
 		{
-			if (type == typeof(sbyte))
-			{
-				((sbyte[])allowedTypes[typeof(sbyte)])[0] = (sbyte)o;
-				return allowedTypes[typeof(sbyte)];
-			}
-			else if (type == typeof(byte))
-			{
-				((byte[])allowedTypes[typeof(byte)])[0] = (byte)o;
-				return allowedTypes[typeof(byte)];
-			}
-			else if (type == typeof(char))
-			{
-				((char[])allowedTypes[typeof(char)])[0] = (char)o;
-				return allowedTypes[typeof(char)];
-			}
-			else if (type == typeof(bool))
+			Type type = typeof(T);
+			if (type == typeof(sbyte) || type == typeof(byte) || type == typeof(char) || type == typeof(bool) || type == typeof(short) || type == typeof(ushort) || type == typeof(int) ||
+				type == typeof(uint) || type == typeof(long) || type == typeof(ulong) || type == typeof(float) || type == typeof(double))
 			{
-				((bool[])allowedTypes[typeof(bool)])[0] = (bool)o;
-				return allowedTypes[typeof(bool)];
-			}
-			else if (type == typeof(short))
-			{
-				((short[])allowedTypes[typeof(short)])[0] = (short)o;
-				return allowedTypes[typeof(short)];
-			}
-			else if (type == typeof(ushort))
-			{
-				((ushort[])allowedTypes[typeof(ushort)])[0] = (ushort)o;
-				return allowedTypes[typeof(ushort)];
-			}
-			else if (type == typeof(int))
-			{
-				((int[])allowedTypes[typeof(int)])[0] = (int)o;
-				return allowedTypes[typeof(int)];
-			}
-			else if (type == typeof(uint))
-			{
-				((uint[])allowedTypes[typeof(uint)])[0] = (uint)o;
-				return allowedTypes[typeof(uint)];
-			}
-			else if (type == typeof(long))
-			{
-				((long[])allowedTypes[typeof(long)])[0] = (long)o;
-				return allowedTypes[typeof(long)];
-			}
-			else if (type == typeof(ulong))
-			{
-				((ulong[])allowedTypes[typeof(ulong)])[0] = (ulong)o;
-				return allowedTypes[typeof(ulong)];
-			}
-			else if (type == typeof(float))
-			{
-				((float[])allowedTypes[typeof(float)])[0] = (float)o;
-				return allowedTypes[typeof(float)];
-			}
-			else if (type == typeof(double))
-			{
-				((double[])allowedTypes[typeof(double)])[0] = (double)o;
-				return allowedTypes[typeof(double)];
+				return new T[1] { (T)o };
```

Comment:
This will cause memory issues as I believe we are caching this by  declaring it above the class as we did previously. We shouldn't constantly return new up the variables. I would instead keep the same logic for variables and just cast it better than we did here.